crates/ruff_python_formatter/src/other/with_item.rs

```diff
@@ -25,25 +24,21 @@ impl FormatNodeRule<WithItem> for FormatWithItem {
         maybe_parenthesize_expression(context_expr, item, Parenthesize::IfRequired).fmt(f)?;
 
         if let Some(optional_vars) = optional_vars {
-            write!(
-                f,
-                [space(), text("as"), trailing_comments(trailing_as_comments)]
-            )?;
-            let leading_var_comments = comments.leading_comments(optional_vars.as_ref());
-            if leading_var_comments.is_empty() {
-                write!(f, [space(), optional_vars.format()])?;
+            write!(f, [space(), text("as"), space()])?;
+
+            if trailing_as_comments.is_empty() {
+                write!(f, [optional_vars.format()])?;
```

Comment:
Nit for consistency (because you don't use `write` in the else branch). I don't feel too opinionated about this. I would also feel fine to always use `write`, because it makes code changes easier

```suggestion
                optional_vars.format().fmt(f)?;
```

Reply:
Good call RE consistency. I find myself preferring `write` for the reason you mention.